src/Tools/dotnet-collect/ConfigPathDetector.cs

```diff
@@ -66,7 +63,10 @@ internal static unsafe string proc_pidpath(int pid)
                 // The path is a fixed buffer size, so use that and trim it after
                 int result = 0;
                 byte* pBuffer = stackalloc byte[PROC_PIDPATHINFO_MAXSIZE];
-                int _pid = getpid();
+
+                // WARNING - Despite its name, don't try to pass in a smaller size than specified by PROC_PIDPATHINFO_MAXSIZE.
+                // For some reason libproc returns -1 if you specify something that's NOT EQUAL to PROC_PIDPATHINFO_MAXSIZE
+                // even if you declare your buffer to be smaller/larger than this size. 
                 result = proc_pidpath(pid, pBuffer, (uint)(PROC_PIDPATHINFO_MAXSIZE * sizeof(byte)));
                 if (result <= 0)
```

Comment:
Even though you didn't add or change the throw below, can you make the same change to use a derived exception like ArgumentException or InvalidOperationException?

Reply:
And change the catch in TryDetectConfigPath to that exception.